src/stream.ts

```diff
@@ -231,10 +252,35 @@ export class WebRTCStream implements Stream {
 
   private _sendFlag(flag: pb.Message_Flag): void {
     try {
-      log('Sending flag: %s', flag.toString());
-      this.channel.send(pb.Message.toBinary({ flag: flag }));
+      log.trace('Sending flag: %s', flag.toString());
+      const msgbuf = pb.Message.toBinary({flag: flag});
+      this.channel.send(lp.encode.single(msgbuf).subarray());
     } catch (e) {
       log.error(`Exception while sending flag ${flag}: ${e}`);
     }
   }
 }
+
+class MessageState {
+  public buffer: Uint8Array = new Uint8Array()
+  public messageSize: number = 0;
+
+  public bytesRemaining(): number {
```

Comment:
I am not a type script expert, but is this not what you would use getters for?, so that they act like regular (read-only) properties?